instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientHttpAttributesExtractor.java

```diff
@@ -68,4 +76,12 @@ protected Long responseContentLengthUncompressed(
       ApacheHttpClientRequest request, HttpResponse response) {
     return null;
   }
+
+  @Override
+  protected List<String> responseHeader(
+      ApacheHttpClientRequest request, HttpResponse response, String name) {
+    return Arrays.stream(response.getHeaders(name))
+        .map(Header::getValue)
+        .collect(Collectors.toList());
```

Comment:
i'm a bit concerned about streams API overhead (memory allocation in particular) for small collections like this

Reply:
I've removed those few streams API calls and replaced them with a simple loop